gateway/tenant.go

```diff
@@ -0,0 +1,62 @@
+package gateway
+
+import (
+	"log"
+	"os"
+
+	"gopkg.in/yaml.v2"
+)
+
+type Tenant struct {
+	ID       string `yaml:"ID"`
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
+}
+
+type config struct {
+	Basic map[string]Tenant `yaml:"basic"`
+}
+
+var tenants = make([]Tenant, 0)
+var isTenantsInitialized = false
+
+func InitTenants(filePath string) {
+	data, err := readYaml(filePath)
+	if err != nil {
+		panic(err)
+	}
+
+	tenantData := data.Basic
+	for key, val := range tenantData {
+		tenants = append(tenants, Tenant{
+			ID:       val.ID,
+			Username: key,
+			Password: val.Password,
+		})
+	}
+	isTenantsInitialized = true
+}
+
+func GetTenants() []Tenant {
+	if !isTenantsInitialized {
+		log.Println("No tenant is provided, returning an empty slice")
```

Comment:
When logging, it's important if we have warning or an error. When we switch the logging library you will have that option.